core/domain/rights_manager.py

```diff
@@ -90,10 +115,13 @@ def _save_activity_rights(
     activity_rights.validate()
 
     if activity_type == constants.ACTIVITY_TYPE_EXPLORATION:
-        model_cls = exp_models.ExplorationRightsModel
+        model_cls: Union[
+            Type[exp_models.ExplorationRightsModel],
+            Type[collection_models.CollectionRightsModel]
+        ] = exp_models.ExplorationRightsModel
     elif activity_type == constants.ACTIVITY_TYPE_COLLECTION:
         model_cls = collection_models.CollectionRightsModel
-    model = model_cls.get(activity_rights.id, strict=False)
+    model = model_cls.get(activity_rights.id, strict=True)
```

Comment:
This seems good.

I wonder -- would it make more sense / be simpler to just define model directly in the if/elif, rather than add an extra model_cls? Also maybe we should add an "else raise Exception" clause to handle the case where activity_type is neither of the two, unless that can be prevented by typing.

Reply:
`model_cls` is removed and now we are directly defining the model.

In the whole file, function `_get_activity_rights` is called before we call `_save_activity_rights` and in `_get_activity_rights` we are already checking if `activity_type` is either of two or not.

https://github.com/oppia/oppia/blob/5f1b2e12f61f3daf750c44cc5dab58071ad582aa/core/domain/rights_manager.py#L471

So, if we add an exception here then I think that expectation will never be executed.

Reply:
Then wouldn't you need to at least assert that activity_type is one of the two possible things? The concern here is if something else calls this function (wrongly) in the future, so an exception still makes sense.

Basically, each function's contract and correctness should be self-contained, don't look at just its current callers. People may modify the code in the future and new (incorrectly-behaving) callers might get introduced.

Reply:
assertion is added